Issue/136 update gradle wrapper

timothyfroehlich opened the pull request:
This change updates the Gradle wrapper to version 8.14.2 to take advantage of the latest features and bug fixes, including Java 24 support.

This also adds the Gradle wrapper validation GitHub action to help ensure wrapper integrity.

Subserial commented:
/gcbrun

Subserial commented:
Thanks for contributing! Could you add a step to the cloud build file to test in Java 24?

timothyfroehlich commented:
> Thanks for contributing! Could you add a step to the cloud build file to test in Java 24?

I haven't been able to test it yet (having Docker headaches) but I think this should work. I went a little beyond your request:

- Also added Java 21 (Google's current JDK standard), just in case
- Changed the image to use straight JDK images instead of Gradle-specific images. This means we won't need to update the image with every Gradle version update, and much more closely aligns with the standard Gradle workflow of using the Gradle wrapper to run the build
- Added a basic "publish" check, (just publishing to a local repo) to make sure the full build/publish tasks are tested.

Subserial commented:
/gcbrun

timothyfroehlich commented:
@Subserial I realized yesterday that I hadn't configured the correct version of the wrapper validation action. I've just pushed an update to use the latest version. Please /gcbrun it and once that's done I'll need you to merge it.

Subserial commented:
/gcbrun
